libd/src/d/semantic/symbol.d

```diff
@@ -754,17 +754,36 @@ struct SymbolAnalyzer {
 			methodIndex = oldMethodIndex;
 		}
 
-		thisType = Type.get(i).getParamType(false, true);
-
+		thisType = Type.get(pass.object.getObject()).getParamType(false, true);
 		import std.conv;
 		auto name = i.name.toString(context);
 		manglePrefix = manglePrefix ~ to!string(name.length);
 
 		i.mangle = "I" ~ manglePrefix;
 
-		assert(d.members.length == 0, "Member support not implemented for interfaces yet");
+		//assert(d.members.length == 0, "Member support not implemented for interfaces yet");
 		assert(d.bases.length == 0, "Interface inheritance not implemented yet");
+		auto members = DeclarationVisitor(pass, AggregateType.Class).flatten(d.members, i);
 
+		foreach(m; members) {
+			if (auto method = cast (Method) m) { // check for (non-static) method
```

Comment:
It doesn't check for non static. Please do not put comment that do not fit the code, that only serve confusion.